### ISSUE TITLE
ArrayInput remove an item on a flat string array

### ISSUE THREAD

**Issue by divanitskiy** [marmelab/react-admin#7124]
**What you were expecting:**
ArrayInput component should remove the exact item a user clicks.

Initial data:
``string_array: ["val1"]``

**What happened instead:**
A wrong item was removed.

**Steps to reproduce:**
Scenario 1.
1. Edit a post. Note that it contains: `string_array: ["val1"]`
2. Add a second item to the `String array` property and leave it empty.
3. Add a third item `val3`.
4. Try removing the first item - the second item is removed instead.

Scenario 2.
1. Edit a post. Note that it contains: `string_array: ["val1"]`
2. Remove the first item of the `String array` property.
3. Add a new item - it should be empty but we get `val1` instead.

**Related code:**

https://codesandbox.io/s/loving-shockley-fbvq2?file=/src/data.tsx

```
<ArrayInput source="string_array">
            <SimpleFormIterator>
              <TextInput source="" fullWidth />
            </SimpleFormIterator>
</ArrayInput>
```

**Other information:**

**Environment**

* React-admin version: 3.19.6
* Last version that did not exhibit the issue (if applicable):
* React version: 17.0.0
* Browser: Chrome
* Stack trace (in case of a JS error):

**Comment by WiXSL** [comment]
The following PR fix it #7123

**Comment by WiXSL** [comment]
Fixed by #7123
